ansible/doc/README.testbed.Internal.md

```diff
@@ -0,0 +1,129 @@
+# Testbed internals
+
+## Topology definition
+
+ - List of currently defined topologies in veos inventory file
+```
+[servers:vars]
+topologies=[‘t1', ‘t1-lag', 't0', 'ptf32', 'ptf64']
+```
+ - Topologies stored inside of vars/topo_*.yml, where * is a topology name
+ - Configuration templates for the topologies saved in roles/eos/templates/*.yml
+
+## Topology file
+
+ - Topology file is a regular ansible yaml file with variables:
+   - topology – defines physical topology
+   - configuration – defines variables for VMs configuration templates
+   - configuration_properties – defines group variables for VMs configuration templates
+
+ - topology dictionary is required
+ - configuration and configuration_properties are optional and used only for topologies with VMs
+
+## Topology file. topology dictionary
+
+ - Two dictionaries:
+   - host_interface – defines a list of port offsets which would be inserted into the PTF container
+   - VMs – defines a list and a physical configuration of VMs used in topology
+
+```
+topology:
+  host_interfaces:
+    - 0
+    - 1
+  VMs:
+    ARISTA01T1:
+      vlans:
+        - 2
+        - 3
+      vm_offset: 0
+    ARISTA02T1:
+      vlans:
+        - 4
+      vm_offset: 1
+
+```
+ - ARISTA01T1 – hostname for a VM
+ - vlans - list of vlan offsets used in VM
+ - vm_offset – offset of VM with base configured as vm_base in testbed.csv
+
+ - In this example:
+   - Let’s consider: vm_base == VM0100, vlan_base == ‘100’
+   - First VM:
+     - hostname ARISTA01T1
+     - Uses VM with physical name VM0100
+     - Ethernet1 is connected to vlan 102
+     - Ethernet2 is connected to vlan 103
+     - Ethernet9 is connected to Ethernet9 of ARISTA02T1
```

Comment:
Need an explanation why? I don't see any '9'-s in the example above. We need a statement that VMs in every topology has a backplane.